pkg/issuer/venafi/client/venaficlient.go

```diff
@@ -70,15 +82,90 @@ type connector interface {
 	RenewCertificate(req *certificate.RenewalRequest) (requestID string, err error)
 }
 
-func New(namespace string, secretsLister corelisters.SecretLister, issuer cmapi.GenericIssuer) (Interface, error) {
-	cfg, err := configForIssuer(issuer, secretsLister, namespace)
+func secretNameForIssuer(venCfg *cmapi.VenafiIssuer) string {
+	switch {
+	case venCfg.TPP != nil:
+		return venCfg.TPP.CredentialsRef.Name
+	case venCfg.Cloud != nil:
+		return venCfg.Cloud.APITokenSecretRef.Name
+	}
```

Comment:
So just to confirm, the new refresh token is being written into the same user-supplied Secret resource? What happens if someone is using a GitOps like controller to create that Secret in the first place? Won't the GitOps controller 'delete' the newly written refresh token, meaning the entire auth flow has to be re-done due to the refresh token being lost?